tests/framework/data/SqlDataProviderTest.php

```diff
@@ -32,7 +32,7 @@ public function testTotalCount()
             'sql' => 'select * from `customer`',
             'db' => $this->getConnection(),
         ]);
-        $this->assertSame(3, $dataProvider->getTotalCount());
+        $this->assertSame(3, (int) $dataProvider->getTotalCount());
```

Comment:
as long as PDO does not care about types we can not do anything useful on Yii side. Once the type is gone that information is lost and we can not guess the original type to cast it.